napari/utils/key_bindings.py

```diff
@@ -459,10 +467,17 @@ def release_key(self, key_combo):
             Key combination.
         """
         key, _ = parse_key_combo(key_combo)
-        try:
-            next(self._key_release_generators[key])  # call function
-        except (KeyError, StopIteration):
-            pass
+        with contextlib.suppress(KeyError, StopIteration):
+            val = self._key_release_generators[key]
+            if isinstance(val, tuple):
+                gen, start = val
+                if (
+                    time.time() - start
+                    > get_settings().application.hold_button_delay
+                ):
+                    gen()
+            else:
+                next(val)  # call function
```

Comment:
The logic here is now a bit more complicated — `gen()` in this case is not a generator, right? So it's probably not the right name? Maybe it is worth adding a comment explaining the two paths. "If it's a tuple, it is a function that will reset the mode back together with a timestamp. Otherwise, it is a generator that..." (and tbh I don't understand the logic at that point 😅)

Reply:
Could you check now?